exportReport/Export/ExportReport.java

```diff
@@ -0,0 +1,255 @@
+				deleteFile(path + fileName + ".xml", "The merged XML file: ");
+				show = true;
+				break;
+			case Utils.bootstrap:
+				view.showMessageDialog("Bootstrap: Currently unavilable, expected release is v2.0.");
+				f_view = null;
+				break;
+			case Utils.xml:
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				show = true;
+				break;
+			case Utils.json:
+				view.showMessageDialog(MessageFormat.format("JSON: File will be generated but won't automatically open.", new Object[] { absolutePath }));
+				f_view = ReportExport.TRANSFORMATION(view, logger, absolutePath, (path + fileName + ".xml"), XML_PATH, mergeXSL);
+				f_view = null;
+				f_view = ReportExport.JSON_EXPORT(view, logger, absolutePath, absolutePath,(path + fileName + ".xml"));
+				deleteFile(path + fileName + ".xml", "The merged XML file: ");
+				break;
+			case Utils.pdf:
+				view.showMessageDialog("PDF: Currently unavilable, expected release is v2.0.");
+				f_view = null;
+				break;
+			case Utils.doc:
+				view.showMessageDialog("DOC: Currently unavilable, expected release is v2.0.");
+				f_view = null;
+				break;
+			default:
+				break;
+			}
+			deleteFile(XML_PATH, "The data dump file: ");
+			try {
+				if (!(f_view == null) && (show == true))
+				{
+					DesktopUtils.openUrlInBrowser(f_view.toURI());					
+				}
+			} catch (Exception e) {
+				logger.error(e.getMessage(), e);
+				view.showMessageDialog(MessageFormat.format("Error opening file.", new Object[] { absolutePath }));
+			}
+			return null;
+		}
+
+		/**
+		 * Executed in event dispatching thread
+		 */
+		@Override
+		public void done() {
+			Toolkit.getDefaultToolkit().beep();
+		}
+	}
+
+	public void generateReport(ViewDelegate view, ExtensionExportReport extension) {
+		FileList list = generateFileList();
+		try {
+			if (fc == null) {
+				fc = generateWriteableFileChooser(fc, list);
+				fc.addPropertyChangeListener(WritableFileChooser.FILE_FILTER_CHANGED_PROPERTY,
+						new PropertyChangeListener() {
+							@Override
+							public void propertyChange(PropertyChangeEvent evt) {
+								ReportFilter filter = (ReportFilter) evt.getNewValue();
+								String extension = (filter.getExtensionByDescription(filter.getDescription())
+										.length() == 0 ? ""
+												: "." + filter.getExtensionByDescription(filter.getDescription()));
+								fc.setSelectedFile(new File(extension));
+							}
+						});
+			}
+
+			int rc = fc.showSaveDialog(View.getSingleton().getMainFrame());
+
+			boolean bool = false;
+			String fileExtension = "";
+			while (rc == WritableFileChooser.APPROVE_OPTION && !bool) {
+				fileExtension = list.compareExtension(fc.getSelectedFile().getName());
+				if (fileExtension.length() > 0) {
+					bool = true;
+				}
+				if (!bool) {
+					JOptionPane.showMessageDialog(null,
+							"The file " + fc.getSelectedFile() + " is not a valid destination file.", "Open Error",
```

Comment:
I think both of those messageDialog options are used in the code base. I'd have to search or defer to @psiinon / @thc202.

As for using an internationalized message you can do something like:
`Constant.messages.getString("extensionname.filechooser.error.destination.invalid", fc.getSelectedFile());`
Then in messages.properties: `extensionname.filechooser.error.destination.invalid=The file {0}  is not a valid destination file.`
(So `fc.getSelectedFile()` will replace `{0}`)

Also if "Open Error" is the title for the dialog it could be internationalized as well.